### PR TITLE
implement benchmarks in ci

### PR DESCRIPTION
**Pull request by Masorubka1**

### CODE REVIEW

**Comment by Masorubka1** [comment]
@geseq add pls:
```
git checkout --orphan gh-pages
git rm -rf .
git commit --allow-empty -m "Initial commit on gh-pages"
git push origin gh-pages
```

**Comment by geseq** [comment]
> @geseq add pls:
> 
> ```
> git checkout --orphan gh-pages
> git rm -rf .
> git commit --allow-empty -m "Initial commit on gh-pages"
> git push origin gh-pages
> ```

done

File: benchmark_test.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+	printResultsWithPercentiles(b, "Cancel Order", totalCancelHist)
+}
+
+func runBenchmarkLatency(b *testing.B) ([]float64, []float64) {
+	seed := time.Now().UnixNano()
+	duration := 30 * time.Second
+	lowerBound := decimal.MustParse("50.0")
+	upperBound := decimal.MustParse("100.0")
+	minSpread := decimal.MustParse("0.25")
+	sched := true
```
Comment (geseq) [inline benchmark_test.go]:
```suggestion
	sched := false
```

This is useful for a targeted benchmark that purely analyzes processing latencies without scheduler cost but it’s probably not very useful as a general comparison.

**Review by geseq — CHANGES_REQUESTED** [review]

File: benchmark_test.go (hunk 1 above)
Comment (Masorubka1) [reply benchmark_test.go]: fixed

**Comment by Masorubka1** [comment]
It seems that only repository owners can run benchmarks due to permission restrictions in GitHub Actions, especially when it comes to posting comments on pull requests from forks. Therefore, we’ll need to manually rerun the benchmark jobs as needed.

Thanks for your understanding!

**Review by geseq — APPROVED** [review]

**Pull request merged**